mmedit/models/backbones/sr_backbones/cain_net.py

```diff
@@ -0,0 +1,386 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.runner import load_checkpoint
+
+from mmedit.models.common import make_layer
+from mmedit.models.registry import BACKBONES
+from mmedit.utils import get_root_logger
+
+
+def pixel_shuffle(x, scale=8, up=True):
+    """Up-scale or down-scale by pixel shuffle.
+
+    Args:
+        x (Tensor): Input tensor.
+        scale (int): Scale factor.
+        up (bool): Up-scale or down-scale.
+
+    Returns:
+        y (Tensor): Output tensor.
+    """
+
+    b, c, h, w = x.shape
+    if up:
+        y = F.pixel_shuffle(x, scale)
+    else:
+        h = int(h / scale)
+        w = int(w / scale)
+        x = x.view(b, c, h, scale, w, scale)
+        shuffle_out = x.permute(0, 1, 3, 5, 2, 4).contiguous()
+        y = shuffle_out.view(b, c * scale * scale, h, w)
+    return y
+
+
+def get_padding_functions(x, padding=7):
+    """Generate padding function.
+
+    tensor --padding_input--> padded tensor
+       ↑                            |
+       ------padding_output----------
+
+    Args:
+        x (Tensor): Input tensor.
+        padding (int): Padding size.
+
+    Returns:
+        padding_input (Function): Padding function.
+        padding_output (Function): Depadding function.
+    """
+
+    h, w = x.shape[-2:]
+    padding_width, padding_height = 0, 0
+    if w != ((w >> padding) << padding):
+        padding_width = (((w >> padding) + 1) << padding) - w
+    if h != ((h >> padding) << padding):
+        padding_height = (((h >> padding) + 1) << padding) - h
+    left, right = padding_width // 2, padding_width - padding_width // 2
+    up, down = padding_height // 2, padding_height - padding_height // 2
+    # print(up, down, left, right)
+    if down >= h or right >= w:
+        function = nn.ReplicationPad2d
+    else:
+        function = nn.ReflectionPad2d
+    padding_input = function(padding=[left, right, up, down])
+    padding_output = function(padding=[0 - left, 0 - right, 0 - up, 0 - down])
+    return padding_input, padding_output
+
+
+class ConvNorm(nn.Module):
```

Comment:
How about `ConvNormWithReflectionPad` ? I think the reflection padding should be mentioned in the class name.